src/test/java/com/checkmarx/flow/cucumber/integration/end2end/genericendtoend/GenericEndToEndSteps.java

```diff
@@ -64,8 +61,53 @@ public void runAsService() {
     }
 
     @And("repository is {word}")
-    public void setRepository(String repository) {
-        this.repository = Repository.setTo(repository, this);
+    public void setRepository(String repositoryName) {
+        this.repository = Repository.setTo(repositoryName, this);
+        readEnvironmentProperties();
+        repository.init(this);
+    }
+
+    private void readEnvironmentProperties(){
+        String upperCaseName = repository.name().toUpperCase();
+        boolean isSca = ScaProperties.CONFIG_PREFIX.equalsIgnoreCase(getEngine());
+        if (
+                System.getenv(upperCaseName + "_HOOK_NAMESPACE") == null ||
+                        System.getenv(upperCaseName + "_HOOK_REPO") == null ||
+                        System.getenv(upperCaseName + "_HOOK_TARGET") == null
```

Comment:
It's better to extract the condition inside `if` into a small boolean method/variable with a clear name (e.g. allEnvVariablesAreDefined).